internal/go_repository_test.go

```diff
@@ -75,3 +81,56 @@ func TestDirectives(t *testing.T) {
 		t.Fatal("Unexpected error:\n", err)
 	}
 }
+
+func TestRepoConfig(t *testing.T) {
+	if err := bazel_testing.RunBazel("build", "@bazel_gazelle_go_repository_config//:all"); err != nil {
+		t.Fatal(err)
+	}
+	outputBase, err := getBazelOutputBase()
+	if err != nil {
+		t.Fatal(err)
+	}
+	outDir := outputBase + "/external/bazel_gazelle_go_repository_config"
+	testtools.CheckFiles(t, outDir, []testtools.FileSpec{
+		{
+			Path: "WORKSPACE",
+			Content: `
+# Code generated by generate_repo_config.go; DO NOT EDIT.
+# gazelle:repo test
+
+go_repository(
+    name = "com_github_apex_log",
+    importpath = "github.com/apex/log",
+)
+
+go_repository(
+    name = "errors_go_git",
+    importpath = "github.com/pkg/errors",
+)
+
+go_repository(
+    name = "errors_go_mod",
+    importpath = "github.com/pkg/errors",
+)
+`,
+		},
+	})
+}
+
+func getBazelOutputBase() (string, error) {
```

Comment:
This really should be done by the `bazel_testing` package with the implementation shared by `RunBazel`. I've opened bazelbuild/rules_go#2189 for more flexibility here. Please add:

```
// TODO(bazelbuild/rules_go#2189): call bazel_testing.BazelOutput once implemented.
```